packages/wonder-blocks-form/src/components/text-area.tsx

```diff
@@ -312,45 +317,45 @@ const styles = StyleSheet.create({
         }px`,
     },
     default: {
-        background: color.white,
-        border: `1px solid ${color.offBlack50}`,
-        color: color.offBlack,
+        background: semanticColor.surface.primary,
+        border: `${border.width.hairline}px solid ${semanticColor.border.strong}`,
+        color: semanticColor.text.primary,
         "::placeholder": {
-            color: color.offBlack64,
+            color: semanticColor.text.secondary,
         },
     },
     defaultFocus: {
         ":focus-visible": {
-            borderColor: color.blue,
-            outline: `1px solid ${color.blue}`,
+            borderColor: semanticColor.border.focus,
+            outline: `${border.width.hairline}px solid ${semanticColor.border.focus}`,
             // Negative outline offset so it focus outline is not cropped off if
             // an ancestor element has overflow: hidden
-            outlineOffset: "-2px",
+            outlineOffset: -2,
         },
     },
     disabled: {
-        background: color.offWhite,
-        border: `1px solid ${color.offBlack16}`,
-        color: color.offBlack64,
+        background: semanticColor.action.disabled.secondary,
+        border: `${border.width.hairline}px solid ${semanticColor.border.primary}`,
+        color: semanticColor.text.secondary,
         "::placeholder": {
-            color: color.offBlack64,
+            color: semanticColor.text.secondary,
         },
         cursor: "not-allowed",
         ":focus-visible": {
-            outline: `2px solid ${color.offBlack32}`,
-            outlineOffset: "-3px",
+            outline: `${border.width.thin}px solid ${semanticColor.action.disabled.default}`,
+            outlineOffset: -3,
         },
     },
     error: {
-        background: color.fadedRed8,
-        border: `1px solid ${color.red}`,
-        color: color.offBlack,
+        background: semanticColor.status.critical.background,
+        border: `${border.width.hairline}px solid ${semanticColor.status.critical.foreground}`,
```

Comment:
Would it make sense to have `semanticColor.status.critical.border` here and in the focus-visible state?

Reply:
As mentioned in other PRs, the goal is to keep `status` as it is. I'll create a ticket to also simplify the `action` tokens once we have something more defined a more stable visual direction for the new experience.